libenkf/include/ert/enkf/site_config.h

```diff
@@ -101,6 +103,11 @@ typedef struct site_config_struct site_config_type;
   mode_t                   site_config_get_umask( const site_config_type * site_config );
 
   site_config_type       * site_config_alloc_empty();
+  site_config_type       * site_config_alloc_default();
+  site_config_type       * site_config_alloc_model_config(const char *);
+
+  config_content_type    * site_config_alloc_content(const site_config_type*, config_parser_type*);
```

Comment:
1. Unfortunately, yes (for now..). Currently, the config_content needs the config_parser provided at construction to be alive until the very end of parsing. Should this be changed so that the config_content owns it config_parser, YES! But this requires some effort (in a different part of the code) and should in my opinion it be done in a separate PR.

2. Yes, everyone parsing the site config file needs exactly this config_content set up. And then it should be done one place and I think that it is natural that it is site_config that knows how a site config file should be parsed.